test/auth/auto_detect_spec.js

```diff
@@ -0,0 +1,53 @@
+/* jshint mocha:true */
+var assert = require('assert');
+var sinon = require('sinon');
+var rewire = require('rewire');
+var autoDetect = rewire('../../lib/auth/auto_detect');
+var NativeFlow = require('../../lib/auth/native_flow');
+var RedirectFlow = require('../../lib/auth/redirect_flow');
+var ChromeExtensionFlow = require('../../lib/auth/chrome_extension_flow');
+
+describe('autoDetect', function() {
+  var sandbox;
+
+  beforeEach(function(){
+    sandbox = sinon.sandbox.create();
+  });
+
+  afterEach(function(){
+    sandbox.restore();
+  });
+
+  it('should detect the Native Flow', function() {
+    var env = {
+      process: {
+        env: sandbox.stub().returns(true)
+      }
+    };
+    var flow = autoDetect(env);
+    assert.equal(flow, NativeFlow);
+  });
+
+  it('should detect the Redirect Flow', function() {
+    var env = {
+      window: {
+        navigator: sandbox.stub().returns({id: true})
```

Comment:
yeah, ditto for `navigator`, I the real JS `window.navigator` is just a property, not a function call.